### PR TITLE
node-builder: introduce SHIM_REDEPLOY_CONFIG

### PR DESCRIPTION
**Pull request by sprt**
###### Merge Checklist  
- [x] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [x] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [x] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [x] genPolicy only: Ensured the tool still builds on Windows
- [x] The `upstream/missing` label (or `upstream/not-needed`) has been set on the PR. 

###### Summary 
See commit.

###### Test Methodology

NOT tested by me yet.

### CODE REVIEW

**Review by danmihai1 — APPROVED** [review]

File: tools/osbuilder/node-builder/azure-linux/package_install.sh [hunk 1]
```diff
@@ -62,6 +63,8 @@ cp -a --backup=numbered src/runtime/data/kata-collect-data.sh "${PREFIX}/${DEBUG
 echo "Installing shim binary and configuration"
 cp -a --backup=numbered src/runtime/containerd-shim-kata-v2 "${PREFIX}/${SHIM_BINARIES_PATH}"/"${SHIM_BINARY_NAME}"
 
-cp -a --backup=numbered src/runtime/config/"${SHIM_CONFIG_FILE_NAME}" "${PREFIX}/${SHIM_CONFIG_PATH}/${SHIM_CONFIG_INST_FILE_NAME}"
+if [ "${SHIM_REDEPLOY_CONFIG}" == "yes" ]; then
```
Comment (ms-mahuber) [inline tools/osbuilder/node-builder/azure-linux/package_install.sh:66]: Fully in line with this PR but just one note: look at "Installing SNP shim debug configuration" - there is another instance where a config gets overwritten :)

Comment (sprt) [reply tools/osbuilder/node-builder/azure-linux/package_install.sh:66]: Line 64? That would be the shim binary that is overwritten, which is expected?

Comment (ms-mahuber) [reply tools/osbuilder/node-builder/azure-linux/package_install.sh:66]: Referring to these lines above:
```
echo "Installing SNP shim debug configuration"
	cp -a --backup=numbered src/runtime/config/"${SHIM_DBG_CONFIG_FILE_NAME}" "${PREFIX}/${SHIM_CONFIG_PATH}"/"${SHIM_DBG_CONFIG_INST_FILE_NAME}"
```

Comment (sprt) [reply tools/osbuilder/node-builder/azure-linux/package_install.sh:66]: Oh! Will fix that.

Comment (sprt) [reply tools/osbuilder/node-builder/azure-linux/package_install.sh:70]: Done

**Review by ms-mahuber — APPROVED** [review]

**Pull request merged**